db_test.go

```diff
@@ -0,0 +1,109 @@
+// qan-api2
+// Copyright (C) 2019 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package main
+
+import (
+	"log"
+	"math"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+	"time"
+
+	_ "github.com/golang-migrate/migrate/database/clickhouse"
+	"github.com/jmoiron/sqlx"
+	_ "github.com/kshvakov/clickhouse"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func setup() *sqlx.DB {
+	cmdStr := `docker exec pmm-clickhouse-test clickhouse client -n --query='DROP DATABASE IF EXISTS pmm_test_parts; CREATE DATABASE pmm_test_parts;'`
+	if out, err := exec.Command("/bin/sh", "-c", cmdStr).Output(); err != nil {
```

Comment:
oh my… Something is seriously wrong if we have to write a test this way. We should fix it after GA.

Reply:
Yes, I agree. But this is fastest in performance and shortest in code.